### PR TITLE
varargs for meshgrid

### LINKED ISSUE #10823: [feature request] torch.meshgrid improvements
- [ ] torch.meshgrid should assert inputs have same type
- [ ] torch.meshgrid should support varargs (by method of python wrapper) to match numpy api: https://docs.scipy.org/doc/numpy/reference/generated/numpy.meshgrid.html

### PR DESCRIPTION
**Pull request by xwfye**
Adds vararg support for meshgrid and adds checks for all the tensor arguments to have the same dtype and device.

Fixes: [#10823](https://github.com/pytorch/pytorch/issues/10823), #11446 

The earlier pull request closed without any changes because I had some rebasing issues, so I made another pull request to close out #10823. Sorry for the inconvenience.

cc: @zou3519

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -8832,6 +8832,10 @@ def test_meshgrid(self):
         self.assertEqual(grid_a.shape, torch.Size([1, 3, 2]))
         self.assertEqual(grid_b.shape, torch.Size([1, 3, 2]))
         self.assertEqual(grid_c.shape, torch.Size([1, 3, 2]))
+        grid_a2, grid_b2, grid_c2 = torch.meshgrid([a, b, c])
```
Comment (ezyang) [inline test/test_torch.py]: I think you meant to make this vararg.

**Review by ezyang — DISMISSED** [review]
test is not a test

File: test/test_torch.py (hunk 1 above)
Comment (upriser7) [reply test/test_torch.py]: the test should be fixed now

**Comment by upriser7** [comment]
test for varargs has been fixed

**Comment by SsnL** [comment]
I believe that this is merged on master

**Pull request closed without merging**